extensions/cornerstone-dicom-rt/src/loadRTStruct.js

```diff
@@ -0,0 +1,217 @@
+import dcmjs from 'dcmjs';
+const { DicomMessage, DicomMetaDictionary } = dcmjs.data;
+const dicomlab2RGB = dcmjs.data.Colors.dicomlab2RGB;
+
+export default async function loadRTStruct(
+  extensionManager,
+  rtStructDisplaySet,
+  referencedDisplaySet,
+  headers
+) {
+  const utilityModule = extensionManager.getModuleEntry(
+    '@ohif/extension-cornerstone.utilityModule.common'
+  );
+
+  const { dicomLoaderService } = utilityModule.exports;
+  const imageIdSopInstanceUidPairs = _getImageIdSopInstanceUidPairsForDisplaySet(
+    referencedDisplaySet
+  );
+
+  // Set here is loading is asynchronous.
+  // If this function throws its set back to false.
+  rtStructDisplaySet.isLoaded = true;
+
+  const segArrayBuffer = await dicomLoaderService.findDicomDataPromise(
```

Comment:
We have already read quite a bit of this, and we have various loaders that will load data asynchronously.  This setup makes things quite slow to load - could we just use the existing header and then load the pixel data in the normal way, or if there are any bulkdata items, then load them using the internal URL as required?  That allows fetching the data much more quickly as it can be fetched and decoded in parallel.

Reply:
Last time I tried it with reading pixelData only it gave me cors errors. We are doing that for the SEG as well. What would be the best way to load the pixelData directly without errors?

Reply:
Use the the image loaders already defined in @cornerstonejs to get the data.   There are existing image loaders which will fetch DICOM data and decode it if necessary.  Those loaders can be enhanced with options to direct fetch the full DICOM if that is a preferred solution, and return just the pixel data as requested.  We probably should have a zoom call to go over some of the options, but ultimately it should make things faster/more efficient to load, and given how slow SEGs are to load, that would be useful.